### PR TITLE
First pass at validating an XML document against the TigerData XSD

### PR DESCRIPTION
**Pull request by hectorcorrea**
Adds a rake task showing how to validate the sample XML document provided by Matt in ticket #896 against the XSD provided on that same issue.

The rake task and the sample XML and XSD files are mostly for demonstration purposes and very likely this code (if useful) will be integrated in to the `app` folder, but for now this is mostly for demo and documentation purposes.

Run with: 
`bundle exec rake xml_schema:validate_example`

In addition included a `td_xml_schema_101.md` file with some general guidance on how to read the TigerData XML Schema.

### CODE REVIEW

File: lib/tasks/xml_schema.rake [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  # Replace the reference to `xml.xsd` from the xsd with a reference to our
+  # local copy of the file, otherwise the validation tends to fail because
+  # Nokogiri cannot load the original `xml.xsd` file from the W3C
+  # source. ¯\_(ツ)_/¯
+  # For more information see: https://stackoverflow.com/a/18527198/446681
+  def use_local_xml_xsd(xsd)
+    local_xml_xsd = Pathname.new("./lib/assets/xml.xsd").realpath.to_s
+    xsd.sub!("https://www.w3.org/2001/xml.xsd", "file://#{local_xml_xsd}")
+  end
```
Comment (hectorcorrea) [inline lib/tasks/xml_schema.rake:37]: `use_local_xml_xsd(xsd)` is a horrible hack but that's the only way I could get Nokogiri to validate the XML against the XSD successfully.

File: lib/assets/xml.xsd
Comment (carolyncole) [inline lib/assets/xml.xsd:1]: @hectorcorrea what is this file?  Is it needed?

**Review by carolyncole — COMMENTED** [review]
This is great work @hectorcorrea!  Just one question.

File: lib/assets/xml.xsd [hunk 2]
```diff
@@ -0,0 +1,287 @@
+<?xml version='1.0'?>
```
Comment (carolyncole) [inline lib/assets/xml.xsd:1]:
```suggestion
<!-- Allows us to utilize Nokogiri for the validation.  For some reason utilizing the copy at  https://www.w3.org/2001/xml.xsd causes issues.  See xml_schema rake task for more information -->
<?xml version='1.0'?>
```

**Review by carolyncole — CHANGES_REQUESTED** [review]
Ok, so maybe we can add a comment,  just to link the two files up...

**Review by carolyncole — APPROVED** [review]
Thanks @hectorcorrea!  My partially caffeinated brain is playing the part of us a year from now...  When we think what is this file and delete it...

**Comment by hectorcorrea** [comment]
@carolyncole agree, this is something we'll forget as soon as *squirrel* :)

**Pull request merged**